src/broadcasting.jl

```diff
@@ -1,19 +1,56 @@
-# See: https://docs.julialang.org/en/v1/manual/interfaces/#man-interfaces-broadcasting-1
+"""
+    NamedDimsStyle{S}
+This is a `BroadcastStyle` for NamedDimsArray's
+It preserves the dimension names.
+`S` should be the `BroadcastStyle` of the wrapped type.
+"""
+struct NamedDimsStyle{S <: BroadcastStyle} <: AbstractArrayStyle{Any} end
+NamedDimsStyle(::S) where {S} = NamedDimsStyle{S}()
+NamedDimsStyle(::S, ::Val{N}) where {S,N} = NamedDimsStyle(S(Val(N)))
+NamedDimsStyle(::Val{N}) where N = NamedDimsStyle{DefaultArrayStyle{N}}()
```

Comment:
We cannot, it is part of the spec for subtyping `AbstractArrayStyle`

Reply:
Where are the `Val` methods used? Can we kill them?

Reply:
Ahhh I see. Well, we'll soon be able to get rid of that :) https://github.com/JuliaLang/julia/pull/31916